homeassistant/components/template/__init__.py

```diff
@@ -29,13 +31,52 @@
 
 from .const import CONF_MAX, CONF_MIN, CONF_STEP, CONF_TRIGGER, DOMAIN, PLATFORMS
 from .coordinator import TriggerUpdateCoordinator
+from .helpers import async_get_blueprints
+from .template_entity import TemplateEntity
 
 _LOGGER = logging.getLogger(__name__)
 DATA_COORDINATORS: HassKey[list[TriggerUpdateCoordinator]] = HassKey(DOMAIN)
 
+DATA_COMPONENT: HassKey[EntityComponent[TemplateEntity]] = HassKey(DOMAIN)
+
+
+@callback
+def templates_with_blueprint(hass: HomeAssistant, blueprint_path: str) -> list[str]:
+    """Return all templates that reference the blueprint."""
+    return [
+        entity_id
+        for platform in async_get_platforms(hass, DOMAIN)
+        for entity_id, template_entity in platform.entities.items()
+        if isinstance(template_entity, TemplateEntity)
+        and template_entity.referenced_blueprint == blueprint_path
+    ]
+
+
+@callback
+def blueprint_in_template(hass: HomeAssistant, entity_id: str) -> str | None:
+    """Return the blueprint the template is based on or None."""
+    for platform in async_get_platforms(hass, DOMAIN):
+        if template_entity := platform.entities.get(entity_id):
+            if isinstance(template_entity, TemplateEntity):
```

Comment:
Combine these two conditions with `and`.

Reply:
I can't use `and` since apparently the walrus does not make the assigned variable available right away in the same expression, but I have combined them in a single `if`

Reply:
It works by using a parenthesis. But the current way works too.